src/helpers/script_helpers.sh

```diff
@@ -117,6 +145,15 @@ function run {
     preconditions "$@"
     steps > "$STEPS_FILE"
     run_steps "$STEPS_FILE" undoable
+
+    # Apply global previous-branch restoration
+    # only if we're at the git root.
+    # Any branch restoration logic for the other
+    # case needs to be handled in a custom way by
+    # the command itself
```

Comment:
Since https://github.com/Originate/git-town/commit/16c074ff7fd47de7040b694b0ba2a3227d52839b we are in the Git root during execution. Can't you run this stuff while being in the git root?